### ISSUE TITLE
GitHub Enterprise (GHE) publishing support

### ISSUE THREAD

**Issue by gerrit** [skpm/skpm#172]
We run an internal GHE instance and would like to benefit from `skpm`'s publishing workflow and baked-in auto-updating. Currently, `skpm publish` doesn't support GHE, in fact a lot of places in the code make the assumption. GHE supports the GitHub REST API, so afaict it should be possible to support internal plugin publication using `skpm` relatively easily.

We're happy to work on the implementation of this if you'd accept such a patch.

## Current state

When running `skpm publish minor --repo-url=https://<GHE-HOSTNAME>/user/repo.git` the output is `✖ ERROR Error: Missing "repository" field in the package.json.` which is pretty unhelpful. Same result if a non-github-dot-com repository field exists in `package.json`.

## Proposed changes

Here's what a path to GHE support might look like:

- [ ] be clearer in help & docs when skpm assumes a public GitHub repo / `github.com` URL and when it can be any instance. "GitHub repo" is ambiguous for GHE users and suggest an internal instance could work
- [ ] improve error message for `skpm publish` when used /w non github.com URL
- [ ] support arbitrary repository URLs, assume it's a GHE instance when it doesn't
- [ ] detect when a `--repo-url` is not on GitHub.com and automatically enable `--skip-registry`
- [ ] fail gracefully if a repository URL happens to not point to any GitHub instance

**Comment by simareeno** [comment]
@mathieudutour is it possible to update plugins on GHE now without publishing appcast.xml and zips on public?

**Comment by mathieudutour** [comment]
No, Sketch needs to be able to access them. Otherwise it would mean building a GHE inside Sketch and I don’t think that’s something we will ever do
